design-documents/graph-ql/coverage/company-credit.md

```diff
@@ -0,0 +1,39 @@
+# Queries
+
+```graphql
+
+###### Begin: Extending existing types ######
+type Company {
+    credit: CompanyCredit! @doc(description: "Company credit balance")
+    credit_history: CompanyCreditHistory! @doc(description: "Company credit operations history")
+}
+###### End: Extending existing types ######
+
+
+###### Begin: Defining new types ######
+type CompanyCreditHistory {
+    items: [CompanyCreditOperation]! @doc(description: "An array of company credit operations")
+    page_info: SearchResultPageInfo! @doc(description: "Metadata for pagination rendering")
+}
+
+type CompanyCreditOperation {
+    date: String! @doc(description: "The date of the company credit operation")
+    # Operation may be of enum type
+    operation: String! @doc(description: "The type of the company credit operation")
```

Comment:
Seems like we can use enum instead of string here:
```
const TYPE_ALLOCATED = 1;
    const TYPE_UPDATED = 2;
    const TYPE_PURCHASED = 3;
    const TYPE_REIMBURSED = 4;
    const TYPE_REFUNDED = 5;
    const TYPE_REVERTED = 6;
```

Reply:
Done